### ISSUE TITLE
我在自己的训练集上训练，loss一开始是几百，训了几个epoch到11.5后不再下降了

### ISSUE THREAD

**Issue by a954217436** [xingyizhou/CenterNet#342]
2019-09-10-10-01: epoch: 1 |loss 13.094866 | hm_loss 10.560472 | wh_loss 21.808166 | off_loss 0.353577 | time 3.866667 | 
2019-09-10-10-04: epoch: 2 |loss 11.584103 | hm_loss 9.208497 | wh_loss 21.272022 | off_loss 0.248404 | time 3.766667 | 
2019-09-10-10-08: epoch: 3 |loss 11.529147 | hm_loss 9.208497 | wh_loss 20.684575 | off_loss 0.252193 | time 3.783333 | 
2019-09-10-10-12: epoch: 4 |loss 11.560646 | hm_loss 9.208497 | wh_loss 20.981068 | off_loss 0.254042 | time 3.783333 | 
2019-09-10-10-16: epoch: 5 |loss 11.544007 | hm_loss 9.208497 | wh_loss 20.859415 | off_loss 0.249568 | time 3.766667 | loss 35.187737 | hm_loss 9.208498 | wh_loss 257.274403 | off_loss 0.251798 | time 0.933333 | 
2019-09-10-10-21: epoch: 6 |loss 11.511237 | hm_loss 9.208497 | wh_loss 20.548204 | off_loss 0.247919 | time 3.783333 | 
2019-09-10-10-24: epoch: 7 |loss 11.539610 | hm_loss 9.208497 | wh_loss 20.811827 | off_loss 0.249930 | time 3.783333 | 

训练指令是
python main.py ctdet --exp_id resdcn_101 --batch_size 16 --master_batch 16 --input_res 512 --num_epochs 10  --lr 0.05 --lr_step 5,8 --gpus 0  --num_workers 8  --save_all --arch resdcn_101

在测试集和训练图片上测试demo.py，都没有任何效果，一个框也没有画出来
有人能帮我看下吗，已经反复研究一天了。。。

**Comment by shuoshuoa** [comment]
配置信息能发出来看一下嘛

**Comment by Beam-wi** [comment]
你改了原代码？ 把学习率设小设到5e-5。我之前也遇到过，我改了原代码，后来换成之前的就可以了，不过这里的学习率普遍很小，要设到-5级以下。现在就是训练集和验证集损失下降正常，但测试集震荡。应该是我们用单卡训练的batch size太小了导致的。如果能解决这个问题就好。

**Comment by a954217436** [comment]
> 你改了原代码？ 把学习率设小设到5e-5。我之前也遇到过，我改了原代码，后来换成之前的就可以了，不过这里的学习率普遍很小，要设到-5级以下。现在就是训练集和验证集损失下降正常，但测试集震荡。应该是我们用单卡训练的batch size太小了导致的。如果能解决这个问题就好。

嗯嗯，我把学习率调低了后，训了一夜，第二天效果很好了

**Comment by niaoyu** [comment]
The same problem for me....The hm loss is always 9.208.
I will try to decrease lr

**Comment by cwjhx** [comment]
> > 你改了原代码？ 把学习率设小设到5e-5。我之前也遇到过，我改了原代码，后来换成之前的就可以了，不过这里的学习率普遍很小，要设到-5级以下。现在就是训练集和验证集损失下降正常，但测试集震荡。应该是我们用单卡训练的batch size太小了导致的。如果能解决这个问题就好。
> 
> 嗯嗯，我把学习率调低了后，训了一夜，第二天效果很好了

你最后的学习率调低到多少了？num_epochs 、lr、 lr_step分别设置的多少？我在自己的数据集上但单GPU训练，batch size设置的是16，num_epochs=230， 、lr = 6.25e-5、 lr_step=180,210，最后的loss如下：
2019-11-11-03-07: epoch: 25 |loss 0.614679 | hm_loss 0.128461 | wh_loss 2.645532 | off_loss 0.221665 | time 1.250000 | loss 0.864196 | hm_loss 0.344149 | wh_loss 2.765474 | off_loss 0.243500 | time 0.000000 | 
2019-11-11-03-08: epoch: 26 |loss 0.667415 | hm_loss 0.191322 | wh_loss 2.535130 | off_loss 0.222580 | time 1.250000 | 
2019-11-11-03-10: epoch: 27 |loss 0.646096 | hm_loss 0.162976 | wh_loss 2.620476 | off_loss 0.221072 | time 1.233333 | 
2019-11-11-03-11: epoch: 28 |loss 0.597778 | hm_loss 0.136945 | wh_loss 2.389694 | off_loss 0.221864 | time 1.250000 | 
2019-11-11-03-12: epoch: 29 |loss 0.613969 | hm_loss 0.156918 | wh_loss 2.394613 | off_loss 0.217589 | time 1.250000 | 
2019-11-11-03-13: epoch: 30 |loss 0.567630 | hm_loss 0.122167 | wh_loss 2.276206 | off_loss 0.217842 | time 1.250000 | loss 1.120765 | hm_loss 0.596342 | wh_loss 2.789291 | off_loss 0.245493 | time 0.000000 | 
2019-11-11-03-15: epoch: 31 |loss 0.547736 | hm_loss 0.114164 | wh_loss 2.149268 | off_loss 0.218645 | time 1.266667 | 
2019-11-11-03-16: epoch: 32 |loss 0.536578 | hm_loss 0.109032 | wh_loss 2.108081 | off_loss 0.216738 | time 1.233333 | 
2019-11-11-03-17: epoch: 33 |loss 0.539025 | hm_loss 0.113937 | wh_loss 2.079801 | off_loss 0.217108 | time 1.233333 | 
2019-11-11-03-18: epoch: 34 |loss 0.526744 | hm_loss 0.105402 | wh_loss 2.048818 | off_loss 0.216461 | time 1.250000 | 
2019-11-11-03-20: epoch: 35 |loss 0.537673 | hm_loss 0.111525 | wh_loss 2.074874 | off_loss 0.218660 | time 1.250000 | loss 0.916743 | hm_loss 0.457140 | wh_loss 2.297050 | off_loss 0.229898 | time 0.000000 | 
2019-11-11-03-21: epoch: 36 |loss 0.555215 | hm_loss 0.130317 | wh_loss 2.084297 | off_loss 0.216469 | time 1.250000 | 
2019-11-11-03-22: epoch: 37 |loss 0.541408 | hm_loss 0.117602 | wh_loss 2.100884 | off_loss 0.213718 | time 1.250000 | 
2019-11-11-03-24: epoch: 38 |loss 0.540437 | hm_loss 0.110341 | wh_loss 2.126666 | off_loss 0.217429 | time 1.250000 | 
2019-11-11-03-25: epoch: 39 |loss 0.530429 | hm_loss 0.111774 | wh_loss 2.022815 | off_loss 0.216374 | time 1.250000 | 
2019-11-11-03-26: epoch: 40 |loss 0.507691 | hm_loss 0.102528 | wh_loss 1.904361 | off_loss 0.214727 | time 1.250000 | loss 0.880712 | hm_loss 0.453848 | wh_loss 1.842190 | off_loss 0.242645 | time 0.000000 | 
2019-11-11-03-27: epoch: 41 |loss 0.593224 | hm_loss 0.167438 | wh_loss 2.105134 | off_loss 0.215272 | time 1.250000 | 
2019-11-11-03-29: epoch: 42 |loss 0.525905 | hm_loss 0.115634 | wh_loss 1.981582 | off_loss 0.212112 | time 1.250000 | 
2019-11-11-03-30: epoch: 43 |loss 0.550699 | hm_loss 0.133409 | wh_loss 2.006866 | off_loss 0.216604 | time 1.233333 | 
2019-11-11-03-31: epoch: 44 |loss 0.481917 | hm_loss 0.091961 | wh_loss 1.807041 | off_loss 0.209252 | time 1.250000 | 
2019-11-11-03-32: epoch: 45 |loss 0.460039 | hm_loss 0.075482 | wh_loss 1.709004 | off_loss 0.213656 | time 1.250000 | loss 0.536977 | hm_loss 0.163832 | wh_loss 1.507026 | off_loss 0.222442 | time 0.000000 | 
2019-11-11-03-34: epoch: 46 |loss 0.467762 | hm_loss 0.087429 | wh_loss 1.697641 | off_loss 0.210568 | time 1.250000 | 
2019-11-11-03-35: epoch: 47 |loss 0.469190 | hm_loss 0.081119 | wh_loss 1.766478 | off_loss 0.211423 | time 1.250000 | 
我在测试集上测试，中心点预测一般是准的，但宽高预测的误差比较大

**Comment by zbl-96** [comment]
beacuse   sigmod clamp out [1e-4 , 1-1e-4  ]   hm_loss=   log(1e-4)*[log(1-1e-4)**alpha]=9.2085

**Comment by Cindy0725** [comment]
> 

你好！我想问一下你是用的原代码的focal loss嘛，原代码里focal loss是求了sum，所以我的heatmap focal loss从7降到了2.多就停住了，这个因为求了sum跟class label的数量和图像的长宽都有关系，我换了dataset （sunrgbd)，不知道focal loss要降到多少才算converge
